docs/stdlib/fts.rst

```diff
@@ -0,0 +1,575 @@
+.. _ref_std_fts:
+
+.. versionadded:: 4.0
+
+================
+Full-text Search
+================
+
+The ``fts`` built-in module contains various tools that enable full-text
+search functionality in EdgeDB.
+
+.. list-table::
+    :class: funcoptable
+
+    * - :eql:type:`fts::Language`
+      - Common languages :eql:type:`enum`
+
+    * - :eql:type:`fts::PGLanguage`
+      - Postgres languages :eql:type:`enum`
+
+    * - :eql:type:`fts::ElasticLanguage`
+      - ElasticSearch languages :eql:type:`enum`
+
+    * - :eql:type:`fts::LuceneLanguage`
+      - Apache Lucene languages :eql:type:`enum`
+
+    * - :eql:type:`fts::Weight`
+      - Weight category :eql:type:`enum`
+
+    * - :eql:type:`fts::document`
+      - Opaque document type
+
+    * - :eql:func:`fts::search`
+      - :eql:func-desc:`fts::search`
+
+    * - :eql:func:`fts::with_options`
+      - :eql:func-desc:`fts::with_options`
+
+When considering FTS functionality our goal was to come up with an interface
+that could support different backend FTS providers. To achieve that we've
```

Comment:
We should mention somewhere that at the moment, EdgeDB supports only postgres FTS backend